src/grasshopper/lib/journeys/base_journey.py

```diff
@@ -112,37 +113,73 @@ def _check_for_threshold_parameters_and_set_thresholds(
     ):
         if parameter_key == "thresholds":
-            for raw_trend_name, threshold_less_than_in_ms in parameter_value.items():
-                trend_name, request_type = self._extract_trend_name(raw_trend_name)
-                thresh_object = {
-                    "less_than_in_ms": threshold_less_than_in_ms,
-                    "actual_value_in_ms": None,
-                    "percentile": 0.8,
-                    "succeeded": None,
-                    "http_method": request_type,
-                }
-                if trend_name in self.environment.stats.trends:
-                    self.environment.stats.trends[trend_name]["thresholds"].append(
-                        thresh_object
-                    )
-                else:
-                    self.environment.stats.trends[trend_name] = {
-                        "tags": self.scenario_args.get("tags", {}),
-                        "thresholds": [thresh_object],
+            thresholds_shape = parameter_value
+            if self._verify_thresholds_shape(thresholds_shape):
+                for trend_name, threshold_values in thresholds_shape.items():
+                    thresh_object = {
+                        "less_than_in_ms": int(threshold_values.get("limit")),
+                        "actual_value_in_ms": None,
+                        "percentile": float(threshold_values.get("percentile", 0.9)),
+                        "succeeded": None,
+                        "http_method": str(threshold_values.get("type")).upper(),
                     }
+                    if trend_name in self.environment.stats.trends:
+                        self.environment.stats.trends[trend_name]["thresholds"].append(
+                            thresh_object
+                        )
+                    else:
+                        self.environment.stats.trends[trend_name] = {
+                            "tags": self.scenario_args.get("tags", {}),
+                            "thresholds": [thresh_object],
+                        }
+            else:
+                logging.error(
+                    "Skipping registering thresholds due to invalid "
+                    "thresholds shape..."
+                )
 
     @staticmethod
-    def _extract_trend_name(raw_trend_name: str):
-        if "{" in raw_trend_name and "}" in raw_trend_name:
-            trend_name = raw_trend_name.split("}")[1]
-            request_type = raw_trend_name.split("}")[0].split("{")[1].upper()
-            return trend_name, request_type
-        else:
-            raise ValueError(
-                "Invalid Trend Name! Please include the request type in curly "
-                + "brackets. E.G: `{POST}post_job_group`, "
-                + "`{CUSTOM}PX_TREND_photon_flow_run`, etc.."
+    def _verify_thresholds_shape(thresholds_shape):
+        valid_types = ["GET", "POST", "PUT", "DELETE", "HEAD", "PATCH", "CUSTOM"]
+        is_valid_shape = True
+        if type(thresholds_shape) != dict:
```

Comment:
probably this is too restrictive. really we could get passed any kind of dict-like object and that would be sufficient. following the python philosophy of duck typing, we don't want to unnecessarily restrict. i might suggest
`isinstance(thresholds_shape,collections.Mapping)`

Reply:
changed to `if not isinstance(thresholds_shape, abc.Mapping):`